src/types/interfaces.ts

```diff
@@ -195,3 +195,64 @@ export type HeadsetControlsRejection = TypedJsonRpcMessage<'headsetControlsRejec
 export type HeadsetControlsChanged = TypedJsonRpcMessage<'headsetControlsChanged', HeadsetControlsChangedParams>;
 
 export type GenesysMediaMessage = HeadsetControlsRequest | HeadsetControlsRejection | HeadsetControlsChanged;
+
+export type FlatObject = {
+  [key: string]: string | number | boolean | null | Date;
+};
+
+export type GenericAction = { _eventType: string; };
+
+export type InsightReport = {
+  appName: string;
+  appVersion: string;
+  originAppName?: string;
+  originAppVersion?: string;
+  actions: InsightAction<any>[];
+};
+
+export type InsightAction<T extends { _eventType: string }> = {
+  actionName: 'WebrtcStats';
+  details: InsightActionDetails<T>;
+};
+
+export type InsightActionDetails<K extends { _eventType: string }> = {
+  _eventType: K['_eventType'];
+  /**
+   * This should be an ISO string
+   */
+  _eventTimestamp: string;
```

Comment:
Ok, so this is the field we will look at nr for the actual time of the event. We can't actually set the one NR relies on by default. NR will always set it to the date the custom event was received (yes, it's stupid).